libraries/mbed/targets/hal/TARGET_Maxim/TARGET_MAX32600/pwmout_api.c

```diff
@@ -77,9 +77,9 @@ void pwmout_init(pwmout_t* obj, PinName pin)
                     } 
 
                     // If all instances are in use, overwrite the last 
-                    pwm = PinMap_PWM[++i];
+                    pwm = PinMap_PWM[i++];
```

Comment:
I looked at  the entire while - while - if code there and it's hard  to understand. Can this be simplified?